src/table/internal.tsx

```diff
@@ -169,255 +222,251 @@ const InternalTable = React.forwardRef(
         const widthsChanged = widthsDetail.some((width, index) => columnDefinitions[index].width !== width);
         if (widthsChanged) {
           fireNonCancelableEvent(onColumnWidthsChange, { widths: widthsDetail });
+          updateCellOffsets();
         }
       },
       singleSelectionHeaderAriaLabel: ariaLabels?.selectionGroupLabel,
       stripedRows,
+      getStickyColumnProperties,
+      stickyColumns,
+      cellOffsets,
+      shouldDisableStickyColumns,
     };
 
-    // Allows keyboard users to scroll horizontally with arrow keys by making the wrapper part of the tab sequence
-    const isWrapperScrollable = tableWidth && containerWidth && tableWidth > containerWidth;
-    const wrapperProps = isWrapperScrollable
-      ? { role: 'region', tabIndex: 0, 'aria-label': ariaLabels?.tableLabel }
-      : {};
-
-    const getMouseDownTarget = useMouseDownTarget();
-    const wrapWithInlineLoadingState = (submitEdit: TableProps['submitEdit']) => {
-      if (!submitEdit) {
-        return undefined;
-      }
-      return async (...args: Parameters<typeof submitEdit>) => {
-        setCurrentEditLoading(true);
-        try {
-          await submitEdit(...args);
-        } finally {
-          setCurrentEditLoading(false);
-        }
-      };
-    };
-
-    const hasDynamicHeight = computedVariant === 'full-page';
-    const overlapElement = useDynamicOverlap({ disabled: !hasDynamicHeight });
-
     useTableFocusNavigation(selectionType, tableRefObject, visibleColumnDefinitions, items?.length);
-
     return (
       <ColumnWidthsProvider
         tableRef={tableRefObject}
         visibleColumnDefinitions={visibleColumnDefinitions}
         resizableColumns={resizableColumns}
         hasSelection={hasSelection}
       >
-        <InternalContainer
-          {...baseProps}
-          __internalRootRef={__internalRootRef}
-          className={clsx(baseProps.className, styles.root)}
-          header={
-            <>
-              {hasHeader && (
-                <div
-                  ref={overlapElement}
-                  className={clsx(hasDynamicHeight && [styles['dark-header'], 'awsui-context-content-header'])}
-                >
-                  <div className={clsx(styles['header-controls'], styles[`variant-${computedVariant}`])}>
-                    <ToolsHeader header={header} filter={filter} pagination={pagination} preferences={preferences} />
+        <StickyColumnsContextProvider
+          wrapperRef={wrapperRefObject}
+          tableRef={tableRefObject}
+          shouldDisableStickyColumns={shouldDisableStickyColumns}
+        >
```

Comment:
How much sticky columns implementation depends on the table implementation? I think it would be great to make this experiment e.g. by creating a test page where sticky columns is added for a simple table instead. That can have the following advantages:

1. Ensure sticky columns feature is properly encapsulated;
2. Make it possible to reuse the feature in case there will be more table implementations (relational table, grouped table etc.);
3. Potential of sharing the util via the toolkit to be used for custom tables.